File: Python/arraysAndHashing/ransomeNote.py
def can_construct(ransomNote: str, magazine: str) -> bool:
    alphabetFrequencyTracker = [0] * 26
    
    for i in magazine:
        alphabetFrequencyTracker[ord(i) - 65] += 1
        
    for j in ransomNote:
        alphabetFrequencyTracker[ord(j) - 65] -= 1
        
    for k in alphabetFrequencyTracker:
        if k < 0: return False
        
    return True

File: Go/Arrays And Hashing/Ransome Note.go
package arraysAndHashing

func CanConstruct(ransomNote string, magazine string) bool {
	alphabetFrequencyTracker := make([]int, 26)

	for _, char := range magazine {
		alphabetFrequencyTracker[char-97]++
	}

	for _, char := range ransomNote {
		alphabetFrequencyTracker[char-97]--
	}

	for _, char := range alphabetFrequencyTracker {
		if char < 0 {
			return false
		}
	}

	return true
}
